ghcide/src/Development/IDE/Core/Compile.hs

```diff
@@ -1455,18 +1455,6 @@ loadInterface session ms linkableNeeded RecompilationInfo{..} = do
 
     case (mb_checked_iface, recomp_iface_reqd) of
       (Just iface, UpToDate) -> do
-         -- If we have an old value, just return it
```

Comment:
It turns out that reusing the same `ModDetails` as we had last time can keep the previous `HscEnv` and corresponding `HomePackageTable` data structures alive. Solution is to delete this code and always regenerate the `ModDetails` from the interface.

Reply:
Yes!! Very important to do this (and cheap)

Reply:
Do we need a cautionary note? should we somehow get rid of `old_value` entirely? or is it okay to use it so long as we don't retain anything from it?

Reply:
We need it for the old `ModIface` (which is safe to use).